### ISSUE TITLE
How to custom baud rate?

### ISSUE THREAD

**Issue by KTOC** [gbmhunter/CppLinuxSerial#7]
Hi,

I want to open the serial port with custom baud rate (256000)

```
serialPortCAN = SerialPort();
serialPortCAN.SetDevice("/dev/ttyUSB0");
serialPortCAN.SetBaudRate(BaudRate::CUSTOM);
```

but it throws exception:

```
Attempting to open COM port "/dev/ttyUSB0".
Configuring COM port "/dev/ttyUSB0".
terminate called after throwing an instance of 'std::runtime_error'
  what():  Custom baud rate not yet supported.
```

How to do this in correct way?

**Comment by UnlightedOtaku** [comment]
you can see this url in SerialPort.cpp
[https://gist.github.com/kennethryerson/f7d1abcf2633b7c03cf0](url)
but I didn't try

**Comment by gbmhunter** [comment]
Hi @KTOC and @UnlightedOtaku ,

Thanks for raising an issue! I hadn't yet added support for a custom baud rate, so I decided to look into it after you asked. I assumed it would be pretty easy, but I was far from right, it turns out I had to re-architect most of the serial port configuration to get this to work. But saying that, hopefully this library now supports custom baud rates as of release https://github.com/gbmhunter/CppLinuxSerial/releases/tag/v2.1.0.

I'm very interested to know how you get on with using this, I have only tested this with an Arduino connected to XUbuntu running inside a VM on a Windows host...I'd love to see some feedback that it works on native Linux machines also :-D.
